src/dreyfus_index.erl

```diff
@@ -214,13 +232,62 @@ handle_info({'DOWN',_,_,Pid,Reason}, #state{
     [gen_server:reply(P, {error, Reason}) || {P, _} <- WaitList],
     {stop, normal, State}.
 
-terminate(_Reason, _State) ->
-    ok.
+terminate(Reason, State) ->
+    case Reason of
+        {shutdown, ddoc_updated} ->
+            send_all(State#state.waiting_list, ddoc_updated),
+            ok;
+        _ ->
+            ok
+    end.
 
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
 % private functions.
+send_all(Waiters, Reply) ->
+    [gen_server:reply(From, Reply) || {From, _} <- Waiters].
+
+check_if_index_is_deleted(DbName, DDocResult, CurrentSig) ->
+    case DDocResult of
+        {not_found, deleted} ->
+            IndicesWithSig = indices_with_sig_from_other_ddocs(DbName, CurrentSig);
```

Comment:
this `indices_with_sig_from_other_ddocs` seems to be very expensive, especially doing exactly same operation for all  N shards. 
I would suggest avoid this check. First, it is expensive. Second, it seems to me that it is not very likely that the customers have exact same indexes for the same db and using them at the same time. Third, even if we close the index used by another ddoc the danger is minimal, if will be reopened with the next query/ken job.

Even if we still want to keep this check it should be initiated from a single process spawn from `dreyfus_index_manager` and them results passed to all concerning `dreyfus_index` processes.

@rnewson @brkolla  What do you think?